### PR TITLE
Changhiskhan/datadiff

### PR DESCRIPTION
**Pull request by changhiskhan**
This is on top of #379 

Ignore the "compute_metric" stuff in this PR just look at the data diff changes

### CODE REVIEW

File: python/lance/tests/util/test_versioning.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 import pytz
 
 import lance
-from lance.util.versioning import get_version_asof
+from lance.util.versioning import get_version_asof, compute_metric, diff, LanceDiff, RowDiff, ColumnDiff
```
Comment (eddyxu) [inline python/lance/tests/util/test_versioning.py:21]: isort.

File: python/lance/util/versioning.py [hunk 2]
```diff
@@ -59,3 +67,140 @@ def get_version_asof(ds: FileSystemDataset, ts: [datetime, pd.Timestamp, str]) -
...
+    for v in versions:
+        if isinstance(v, dict):
+            v = v["version"]
+        vdf = metric_func(lance.dataset(uri, version=v))
+        vdf["version"] = v
+        results.append(vdf)
+    return pd.concat(results)
+
+
+def diff(uri, start_version: int, end_version: int) -> LanceDiff:
```
Comment (eddyxu) [inline python/lance/util/versioning.py:91]: should we change the parameters to `v1` and `v2`?

`start/end` makes me think this is find all versions from v1 to v2. like `git log v1..v2`

**Review by eddyxu — APPROVED** [review]
lgtm overall . small qq.

File: python/lance/util/versioning.py (hunk 2 above)
Comment (changhiskhan) [reply python/lance/util/versioning.py]: sure. changed

File: python/lance/tests/util/test_versioning.py (hunk 1 above)
Comment (changhiskhan) [reply python/lance/tests/util/test_versioning.py]: done

**Pull request merged**